TempMail/utils/Email.swift

```diff
@@ -81,12 +71,12 @@ class Email : ObservableObject{
     
     //  getter for email addr
     func getEmailAddr() -> String {
-        return email_addr
+        return email_addr!
     }
     
     //  getter for emails
     func getInbox() -> [InboxModel] {
-        return inbox
+        return inbox ?? [InboxModel]()
```

Comment:
don't need this check here, just return [InboxModel]?

Reply:
I believe you do need this check. Since the inbox is an optional, it can be nil. It is assigned nil until the user gets a mail. If I set it to return [InboxModel]?, then there will be an issue in line 49 of ContentView when we call getInbox, where the value of the optional type must be unwrapped. If nil is unwrapped, then there should be a default value, which I've set as an empty array. else if List(self.email.getInbox()) is force-unwrapped and the value is nil, the execution of the program will stop.

Reply:
The check for nil needs to be changed on the gui side, I'll make the change. As it is, there's no functional difference between nil and [], they're basically placeholders for each other. Conceptually if an inbox has no messages (ie []) then it is nil and vice versa. Since we have optionals, we should use them as intended instead of reverting to familiar behavior. It would be nicer to not show the list of messages if there are no emails in your inbox similar to how the mail app does it. The gui would check if the value is nil then show nothing else show the list of messages.

Reply:
Ok that sounds good. Maybe the gui can check for nil, and if nil, show a grayed out background message along the lines of "No emails currently" or "Mailbox is empty"